advanced/stack_overflow_detection/src/main.rs

```diff
@@ -0,0 +1,59 @@
+#![no_std]
+#![no_main]
+
+use esp_backtrace as _;
+use esp_println::println;
+use hal::{
+    assist_debug::DebugAssist, clock::ClockControl, peripherals::Peripherals, prelude::*,
+    timer::TimerGroup, Rtc,
+};
+
+#[entry]
+fn main() -> ! {
+    let peripherals = Peripherals::take();
+    let system = peripherals.SYSTEM.split();
+    let clocks = ClockControl::boot_defaults(system.clock_control).freeze();
+    let mut peripheral_clock_control = system.peripheral_clock_control;
+
+    // Disable the RTC and TIMG watchdog timers
+    let mut rtc = Rtc::new(peripherals.RTC_CNTL);
+    let timer_group0 = TimerGroup::new(peripherals.TIMG0, &clocks, &mut peripheral_clock_control);
+    let mut wdt0 = timer_group0.wdt;
+    let timer_group1 = TimerGroup::new(peripherals.TIMG1, &clocks, &mut peripheral_clock_control);
+    let mut wdt1 = timer_group1.wdt;
+
+    rtc.swd.disable();
+    rtc.rwdt.disable();
+    wdt0.disable();
+    wdt1.disable();
```

Comment:
Same here

Reply:
Ah you are right! I'm not used to this yet 😄